addons/website/controllers/main.py

```diff
@@ -227,6 +230,10 @@ def sitemap_xml_index(self, **kwargs):
         View = request.env['ir.ui.view'].sudo()
         mimetype = 'application/xml;charset=utf-8'
         content = None
+        url_root = request.httprequest.url_root
+        # cache per domain
+        hashed_url_root = str(abs(hash(url_root)))[:8]
```

Comment:
Any reason for the `abs` and `[:8]`? Nitpicking but it seems to only increase the chance of collision of the hashing at the cost of extra code, no? :grimacing:

Reply:
For `abs` this is because `hash()` can return negative integer:
```py
>>> hash('https://github.com/')
-6983552336004323597
>>> hash('https://odoo.com/')
2152249480296729207
```

For the `[:8]`, that's just to keep it short, risk of collision will be close to null even with 8 first chars but also it it were to collide it means it would just serve the sitemap of the other domain, which is basically the current behavior before this PR.

There might be a better way to generate an unique string based on the domain tho, I just did not really think it more than that and went with the easy way (built-in `hash()`).

Reply:
> For abs this is because hash() can return negative integer

I assure you that I guessed that much :joy: My point was that this double the risk of collision for more code :shrug: But I guess if you really want to avoid the `sitemap-1--329032` format... could make sense, but I would not care IMO :shrug: 

> For the [:8], that's just to keep it short, risk of collision will be close to null even with 8 first chars but also it it were to collide it means it would just serve the sitemap of the other domain, which is basically the current behavior before this PR.

Yes ok, that's why I said "nitpicking", but it's basically more code for worse anyway, no?

> I just did not really think it more than that and went with the easy way (built-in hash()).

That's actually my point: you _did think_ more than that. "Just use `hash`" would be enough and the code would be easier to read (but again, nitpicking) :wink: 

But I'll let you choose then, just confirm me that you don't agree :+1:

Reply:
It is mainly done since it's something visible in the sitemap (if in "index" / split mode), not just for a reference for orm searches.
So instead of
![image](https://github.com/odoo/odoo/assets/30048408/c2551eb0-4055-4af3-8929-cbb44a84b62a)
We have this
![image](https://github.com/odoo/odoo/assets/30048408/388980ab-d30d-4123-9dbd-b2a2f5527d8a)
(No double dashes + shorter)

But it's not something supposed to be seen by humans anyway, and it's only for big sitemap (which turns into split / index sitemap), so I can totally remove it if it bugs you :+1:

Reply:
I'll go with `sha`, after discussion with JKE `hash` has not a fixed seed :facepalm:

Reply:
Changed, using `md5(url_root.encode()).hexdigest()[:8]`
```xml
<loc>http://127.0.0.1:8069/sitemap-1-dcb316b4-4.xml</loc>
```

Reply:
I really would just take all characters of the md5 still but, ok ^^ Thanks @JKE-be for noticing :scream: